### PR TITLE
Improve plot settings ui

### PR DESCRIPTION
**Pull request by Falon452**
1. Seperate code into smaller functions
2. Refresh calculating packets on every 25th update of frame (laggy on large data sets)
3. Fix stacking packets - here was mistake I made - I should have taken previous packet rather than last
image

After changes: 

image

4. Fix plot not filling entire space by removing `.view_aspect(2.0);`
Pasted Graphic

After 

image

5. Added `Alias` column on RtpStreamsTable and possibility to locally change it
6. Added locally filtering streams on streams table and streams plot

Stream alias
value • RTP timestamp Packet timestamp Sequence number
RTP streams
X axis value • RTP timestamp ® Packet timestamp • Sequence number
image
Pasted Graphic 1

### CODE REVIEW

File: client/src/app/gui.rs [hunk 1]
```diff
@@ -95,14 +97,27 @@ impl Gui {
...
+        // For large amount data plot can be laggy, so we don't refresh on every ui update
+        if self.update_counter == u32::MAX {
+            self.update_counter = 0
+        }
+
+        let should_refresh = self.update_counter % 25 == 0;
+
+        self.update_counter += 1;
+        should_refresh
+    }
```
Comment (LVala) [inline client/src/app/gui.rs:119]: I don't like that, especially considering that (I'm pretty sure) egui doesn't actually run these on every frame.

What about something like this: in `rtp_streams_plot.rs` you check if number of packets in stream changed from previous frame, and if it did, you refresh the points. Then there's not need for this frame counting.

You can test it, there probably will be a bit of a lag at the beginning when a lot of new packets is coming, but in general I hope it's gonna be alright. If not, for now we can stick with frame counting.

File: client/src/app/gui/rtp_streams_plot.rs [hunk 2]
```diff
@@ -46,145 +52,264 @@ impl RtpStreamsPlot {
...
+                    stream.display_name.to_string(),
+                    self.settings_x_axis,
+                    &mut self.points_data,
+                );
+            });
+    }
+
+    fn stream_is_hidden(streams_visibility: &HashMap<u32, bool>, ssrc: &u32) -> bool {
+        !(*streams_visibility.get(ssrc).unwrap())
+    }
```
Comment (LVala) [inline client/src/app/gui/rtp_streams_plot.rs:180]: I general, I think that creating private associated methods is kinda pointless when you can put it outside of the `impl` block. I don't see any benefits, on the other hand you have to call with the `Self::something` and they're harder to test, but maybe I'm missing something.

**Review by LVala — CHANGES_REQUESTED** [review]

File: client/src/streams/stream.rs [hunk 3]
```diff
@@ -98,8 +98,8 @@ impl Stream {
                     - self.previous_rtp_timestamp.unwrap() * unit_timestamp;
                 let d = arrival_time_difference.as_secs_f64() - timestamp_difference;
 
-                self.jitter = self.jitter + (d - self.jitter) / 16.0;
-                self.jitter_history.push(self.jitter);
+                self.jitter_in_ms = self.jitter_in_ms + (d - self.jitter_in_ms) / 16.0;
+                self.jitter_history.push(self.jitter_in_ms * 1000.0);
```
[the remark this answers is not in the document]
Comment (Falon452) [reply client/src/streams/stream.rs:102]: good catch

[the remark this answers is not in the document]
Comment (Falon452) [reply client/src/streams/stream.rs:102]: changed to:
```
                self.jitter_in_ms = self.jitter_in_ms + (d - self.jitter_in_ms) / 16.0  * 1000.0;
                self.jitter_history.push(self.jitter_in_ms);
```

File: client/src/app/gui/rtp_streams_plot.rs (hunk 2 above)
Comment (Falon452) [reply client/src/app/gui/rtp_streams_plot.rs]: Agree, moved outside of class all those methods

File: client/src/app/gui.rs (hunk 1 above)
Comment (Falon452) [reply client/src/app/gui.rs]: love the idea

File: client/src/app/gui.rs [hunk 4]
```diff
@@ -63,6 +63,7 @@ pub struct Gui {
     rtp_packets_table: RtpPacketsTable,
     rtp_streams_table: RtpStreamsTable,
     rtp_streams_plot: RtpStreamsPlot,
+    update_counter: u32,
```
Comment (LVala) [inline client/src/app/gui.rs:66]: Is this `update_counter` used for anything or you just forgot to remove it with the "refresh every x frames" function? It's used in some other places as well.

File: client/src/app/gui/rtp_streams_plot.rs [hunk 5]
```diff
@@ -37,154 +42,253 @@ impl Display for SettingsXAxis {
...
+        ui.vertical(|ui| {
+            ui.add_space(10.0);
+        });
+    }
+
+    fn plot_ui(&mut self, ui: &mut Ui) {
+        let number_of_packets = self.streams.borrow().packets.len();
+        if self.last_packets_len != number_of_packets || self.requires_reset {
+            self.refresh_points();
+        }
```
Comment (LVala) [inline client/src/app/gui/rtp_streams_plot.rs:111]: I'm wondering if we shouldn't base this condition on number of RTP packets specifically, not all packets? Imagine that I used "parse as RTP" option on some packet that wasn't recognized as RTP. Number of packets is the same, but number of RTP packets changed. That would require filtering the packets list on every frame, which also kinda sucks.

I think we can leave it as it is, but add a `FIXME:` comment with what I wrote above, more or less.

**Review by LVala — APPROVED** [review]

File: client/src/app/gui/rtp_streams_plot.rs (hunk 5 above)
Comment (Falon452) [reply client/src/app/gui/rtp_streams_plot.rs:111]:
```rust
    fn number_of_rtp_packets(&mut self) -> usize {
        self
            .streams
            .borrow()
            .streams
            .values()
            .map(|stream| stream.rtp_packets.len())
            .sum()
    }
```

Comment (Falon452) [reply client/src/app/gui/rtp_streams_plot.rs:111]: Why wait? LET'S DO IT!

File: client/src/app/gui.rs (hunk 4 above)
Comment (Falon452) [reply client/src/app/gui.rs:66]: thanks 
image

**Pull request merged**